### ISSUE TITLE
Redesign automl recipe

### ISSUE THREAD

**Issue by shanyu-sys** [intel/analytics-zoo#322]
The future usage of recipe might be
```python
# user customized model
def model_creator(config):
    model = tf.keras.models.Sequential([tf.keras.layers.Dense(config["hidden_size"])])
    model.compile()
    return model

# user customized search space corresponding to model hyper parameters
from zoo.orca.automl import hp
search_space = {
    # model parameters
    "hidden_size": hp.grid_search([32, 64, 128]),

    # optimization parameters
    "lr": hp.uniform(0.001, 0.01),
    "batch_size": hp.choice([32, 64, 128])}

# build auto estimator and fit
auto_est = AutoEstimator.from_keras(model_creator=model_creator,...)
auto_est.fit(data, 
             search_space=search_space,
             metric="mse",
             num_samples=1,
             early_stop={"max_iter": 5, "threshold": 0.01}, # same as stop in tune.run(),
             search_alg=,

            )
